tests/components/airtouch5/test_cover.py

```diff
@@ -0,0 +1,147 @@
+        blocking=True,
+    )
+    mock_airtouch5_client.send_packet.assert_called_once()
+    mock_airtouch5_client.reset_mock()
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_OPEN_COVER,
+        {ATTR_ENTITY_ID: COVER_ENTITY_ID},
+        blocking=True,
+    )
+    mock_airtouch5_client.send_packet.assert_called_once()
+    mock_airtouch5_client.reset_mock()
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_SET_COVER_POSITION,
+        {ATTR_ENTITY_ID: COVER_ENTITY_ID, ATTR_POSITION: 50},
+        blocking=True,
+    )
+    mock_airtouch5_client.send_packet.assert_called_once()
+    mock_airtouch5_client.reset_mock()
+
+
+async def test_cover_callbacks(
+    hass: HomeAssistant,
+    mock_airtouch5_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test the callbacks of the Airtouch5 covers."""
+
+    await setup_integration(hass, mock_config_entry)
+
+    # We find the callback method on the mock client
+    zone_status_callback: Callable[[dict[int, ZoneStatusZone]], None] = (
+        mock_airtouch5_client.zone_status_callbacks[2]
+    )
+
+    # Define a method to simply call it
+    async def _call_zone_status_callback(open_percentage: int) -> None:
+        zsz = ZoneStatusZone(
+            zone_power_state=ZonePowerState.ON,
+            zone_number=1,
+            control_method=ControlMethod.PERCENTAGE_CONTROL,
+            open_percentage=open_percentage,
+            set_point=None,
+            has_sensor=False,
+            temperature=None,
+            spill_active=False,
+            is_low_battery=False,
+        )
+        zone_status_callback({1: zsz})
+        await hass.async_block_till_done()
+
+    # And call it to effectively launch the callback as the server would do
+
+    # Partly open
+    open_percentage = 70
+    await _call_zone_status_callback(open_percentage)
+    state = hass.states.get(COVER_ENTITY_ID)
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == open_percentage * 100.0
+
+    # Fully open
+    open_percentage = 100
+    await _call_zone_status_callback(open_percentage)
+    state = hass.states.get(COVER_ENTITY_ID)
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == open_percentage * 100.0
+
+    # Fully closed
+    open_percentage = 0
+    await _call_zone_status_callback(open_percentage)
+    state = hass.states.get(COVER_ENTITY_ID)
+    assert state
+    assert state.state == STATE_CLOSED
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == open_percentage * 100.0
+
+    # Partly reopened
+    open_percentage = 30
+    await _call_zone_status_callback(open_percentage)
+    state = hass.states.get(COVER_ENTITY_ID)
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == open_percentage * 100.0
```

Comment:
Doesn't this mean the position is asserted to be 3000?

Reply:
Yes. I don't know why it was multiplied by 100 in the first place. If it seems out of place for HA, I'll look to fix it in the cover.

Reply:
Correct me if I'm wrong, but isn't the value usually 1-100?

Reply:
I think that's how it should be done yes

Reply:
No worries, will do 🙂

Reply:
It was just an issue with my understanding the backend api. It wanted 0.01 -> 1.0 as it's values. I've corrected the tests to match.